### PR TITLE
Feat/setup gh actions

### PR DESCRIPTION
**Pull request by RMcNeely**
Adds deployment action.

A bit more text to pad everything out some more as well

### CODE REVIEW

File: .github/workflows/deploy-to-gh-pages.yaml [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+          contents: write
+        concurrency:
+          group: ${{ github.workflow }}-${{ github.ref }}
+        steps:
+          - uses: actions/checkout@v3
+
+          - name: Setup Node
+            uses: actions/setup-node@v4
+            with:
+              node-version: "20"
```
Comment (CodeMan99) [inline .github/workflows/deploy-to-gh-pages.yaml:23]: Devcontainer is using node v22.

The two configurations should match. Both be v20 or v22.

File: .github/workflows/deploy-to-gh-pages.yaml [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+          #     path: ~/.npm
+          #     key: ${{ runner.os }}-node-${{ hashFiles('**/package.json') }}
+
+          - name: Persist Eleventy .cache
+            uses: actions/cache@v3
+            with:
+              path: ./.cache
+              key: ${{ runner.os }}-eleventy-fetch-cache
+
+          - run: npm install
```
Comment (CodeMan99) [inline .github/workflows/deploy-to-gh-pages.yaml]:
```suggestion
          - run: npm ci
```

File: .github/workflows/deploy-to-gh-pages.yaml [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+          - run: npm run build
+
+          - name: Deploy
+            uses: peaceiris/actions-gh-pages@v3
```
Comment (CodeMan99) [inline .github/workflows/deploy-to-gh-pages.yaml]: Any reason in particular for using a third-party action here?

Github provides several now: [Deploy](https://github.com/marketplace/actions/deploy-github-pages-site), [Configure](https://github.com/marketplace/actions/configure-github-pages), and [Upload Artifact](https://github.com/marketplace/actions/upload-github-pages-artifact).

Comment (RMcNeely) [reply .github/workflows/deploy-to-gh-pages.yaml]: only that this is adapted right from the 11ty docs and thats what they have. I believe that this documentation may have been contributed by them?

Comment (CodeMan99) [reply .github/workflows/deploy-to-gh-pages.yaml]: Alright, let's run with it, but plan to move to the official actions in short order.

File: .github/workflows/deploy-to-gh-pages.yaml (hunk 1 above)
Comment (RMcNeely) [reply .github/workflows/deploy-to-gh-pages.yaml:23]: Can the devcontainer pull from a .nvmrc file?

Comment (CodeMan99) [reply .github/workflows/deploy-to-gh-pages.yaml:23]: Probably? NVM is setup in the image...

I don't really have much experience with NVM.

Comment (CodeMan99) [reply .github/workflows/deploy-to-gh-pages.yaml:23]: The documentation is here: https://github.com/devcontainers/features/tree/main/src/node

Yes, that is a feature and not an image. The image is just the [base image](https://github.com/devcontainers/images/tree/main/src/base-debian) :heavy_plus_sign: the above `node` feature with defaults... resulting in this [javascript-node](https://github.com/devcontainers/images/tree/main/src/javascript-node) image.

File: .github/workflows/deploy-to-gh-pages.yaml (hunk 3 above)
Comment (RMcNeely) [reply .github/workflows/deploy-to-gh-pages.yaml]: So it looks like [Configure](https://github.com/actions/configure-pages) only works with Next, Nuxt, Gatsby,  and SvelteKit if you dig into the code [here](https://github.com/actions/configure-pages/blob/main/src/set-pages-config.test.js).

**Review by CodeMan99 — COMMENTED** [review]
RE:

> See if upload is actually needed?

Answer: Yes

image

**Review by CodeMan99 — APPROVED** [review]

**Pull request merged**